### ISSUE TITLE
Easiest way to increase mosaic detection sensitivity?

### ISSUE THREAD

**Issue by wheezy1749** [ladaapp/lada#23]
Liking the latest changes. I've gotten lost in training datasets so haven't had much time to mess with the code. You increased performance a lot. The restoration is really good. Much better than deep mosaics. However, the biggest issue seems to be with having portions of the video that are just showing mosaics and not being cleaned. I'm not sure if its because they are not detected, or if they are not cleaned properly. From what I've tested it seems like small mosaics are popping in and out a lot and it when the "scenes" are not continuous it seems to just not clean anything (even though its detected). Not something I've had an issue with on DeepMosaics on the same video. Though the restoration is much better on Lada (and performance).

This is usually on further away shots but the final video is distracting having the mosaics pop in and out. Wondering what you think is going on. Thanks.

**Comment by ladaapp** [comment]
there is no sensitivity or other knobs to adjust mosaic restoration. only the sensitivity for the mosaic detection can be set. so if you say that the mosaic got detected but the restoration did not do anything then there is not much you can do unfortunately.

maybe its related to the training settings or the dataset of the mosaic restoration model.

i think i've used worse degradations like compression artifacts on for the mosaic detection dataset compared to the restoration ds and also did not filter out that many extremely low quality videos/images. if you notice this issue mostly on scenes where quality is pretty bad but not in high quality sources than this could be a reason.

if it happens only on very small mosaic regions but not on bigger mosaics then it could be a similar issue as the mosaic scene is transformed into the target dimensions the restoration model expects [crop, pad, resize into dim 256x256px]. maybe such tiny scenes were mostly filtered out in the restoration model dataset creation process.

if you can confirm this and you don't notice detected but not restored mosaics on high quality content then i could check dataset creation process and my source material...

if you say you didn't notice in in deepmosaics do you mean using the deepmosaic restoration model within lada or running the original deepmosaics source? when you use deepmosaics within lada then it will use ladas mosaic detection model and pre/post processing.

**Comment by ladaapp** [comment]
@ratbunch 

i also did some comparisons and indeed deepmosaics doesn't seem to have this issue. i've tweaked the degradation process for the mosaic restoration dataset and saw some improvements after retraining the model.

can you give this new run a try https://github.com/ladaapp/lada/releases/tag/v0.5.1-beta2
